oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java

```diff
@@ -1693,6 +1694,136 @@ void shouldErrorIfIdTokenHasIncorrectClient() throws JOSEException {
+                        browserSessionIdCookieFromResponse(response));
+            }
+
+            @Test
+            void noExistingSessionAndBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response =
+                        setupWithExistingSession(null, "id-from-cookie");
+
+                verify(sessionService).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("a-new-session-id", sessionCaptor.getValue().getSessionId());
+                assertEquals(
+                        "a-new-browser-session-id", sessionCaptor.getValue().getBrowserSessionId());
+                assertEquals(
+                        "browser-session-id=a-new-browser-session-id; Domain=auth.ida.digital.cabinet-office.gov.uk; Secure; HttpOnly;",
+                        browserSessionIdCookieFromResponse(response));
+            }
+
+            @Test
+            void existingSessionAndNoBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response =
+                        setupWithExistingSession(
+                                new Session("existing-session-id").withBrowserSessionId("same-id"),
+                                null);
+
+                verify(sessionService, never()).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("existing-session-id", sessionCaptor.getValue().getSessionId());
+                assertEquals("same-id", sessionCaptor.getValue().getBrowserSessionId());
+                assertEquals(
+                        "browser-session-id=same-id; Domain=auth.ida.digital.cabinet-office.gov.uk; Secure; HttpOnly;",
+                        browserSessionIdCookieFromResponse(response));
+            }
+
+            @Test
+            void existingSessionWithNoBrowserSessionIdAndBrowserSessionIdInHeaders() {
+                ArgumentCaptor<Session> sessionCaptor = ArgumentCaptor.forClass(Session.class);
+                APIGatewayProxyResponseEvent response =
+                        setupWithExistingSession(
+                                new Session("existing-session-id").withBrowserSessionId(null),
+                                "id-from-cookie");
+
+                verify(sessionService, never()).createSession();
+                verify(sessionService).save(sessionCaptor.capture());
+                assertEquals("existing-session-id", sessionCaptor.getValue().getSessionId());
+                assertNull(sessionCaptor.getValue().getBrowserSessionId());
+                assertEquals(
+                        2, response.getMultiValueHeaders().get(ResponseHeaders.SET_COOKIE).size());
```

Comment:
```suggestion
                assertTrue(
                        response.getMultiValueHeaders().get(ResponseHeaders.SET_COOKIE).stream()
                                .noneMatch(it -> it.startsWith("browser-session-id=")));
```

Reply:
We want a much stronger assertion here, and one which won't break the next time we touch the cookie code